js/src/lib/components/InferenceWidget/shared/helpers.ts

```diff
@@ -52,16 +52,42 @@ export async function getBlobFromUrl(url: string): Promise<Blob> {
 	return blob;
 }
 
-async function callApi(
+interface Success<T> {
+	computeTime: string;
+	output:      T;
+	outputJson:  string;
+	response:    Response;
+	status:      "success";
+}
+
+interface LoadingModel {
+	error:         string;
+	estimatedTime: number;
+	status:        "loading-model";
+}
+
+interface Error {
+	error:  string;
+	status: "error";
+}
+
+interface CacheNotFound {
+	status: "cache not found";
+}
+
+type Result<T> = Success<T> | LoadingModel | Error | CacheNotFound;
+
+export async function getResponse<T>(
```

Comment:
Nitpicking, but `getResponse` is a very generic name

Reply:
maybe `callInferenceApi` ?

Reply:
handled in https://github.com/huggingface/hub-docs/pull/1034/commits/e16171ce99dfce6014a1557a17c77e98f168a203